src/Common/Uuid/UuidMapping.php

```diff
@@ -39,10 +49,49 @@ public static function getMappingForUUID($uuid, $is_binary = false)
         }
         return $result;
     }
+
+    public static function createAllMissingResourceUuids()
+    {
+        // Update for mapped uuids
+        $mappedCounter = 0;
+        $mappedTables = self::UUID_MAPPING_DEFINITIONS;
+        foreach ($mappedTables as $mapping) {
+            if (empty($mapping['codes'])) {
+                $count = self::createMissingResourceUuids($mapping['resource'], $mapping['table']);
+                $mappedCounter += $count;
+            } else {
+                foreach ($mapping['codes'] as $code) {
+                    $count = self::createMissingResourceUuids($mapping['resource'], $mapping['table'], "category=" . $mapping['category'] . "&code=" . $code['code']);
+                    $mappedCounter += $count;
+                }
+            }
+        }
+        return $mappedCounter;
+    }
+
     // For now, support one to one uuid to target table, but will plan to add one uuid to many targets tables in future
     //   when presented with that use case (this is why the uuid column in uuid_mapping is not unique btw).
     public static function createMissingResourceUuids($resource, $table, $resourcePath = null)
     {
+        try {
+            sqlBeginTrans();
+            $counter = 0;
+            do {
+                $count = self::createMissingResourceUuidsStep($resource, $table, $resourcePath);
+                $counter += $count;
+            } while ($count > 0);
+            sqlCommitTrans();
+            return $counter;
+        } catch (Exception $exception) {
```

Comment:
@bradymiller Make sure to add a ```use \Exception;``` clause to the file.  Otherwise if an exception is thrown here it will fatal error here as it can't find Exception in the OpenEMR\Common\Uuid namespace.  I ran into this problem somewhere else.  Not sure why PHPStorm doesn't auto-import / detect this problem.

Reply:
done